### PR TITLE
fixing 500 bug on empty lexical query

### PR DESCRIPTION
**Pull request by RaynorChavez**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
When a user requests a lexical search with `q=""` (an empty string), marqo returns a 500 error.

* **What is the new behavior (if this is a feature change)?**
The bug is due to `_lexical_search` not creating a valid yql expression when the string is empty. We add two new conditionals in `_to_vespa_lexical_query` 

   q="" : Empty results regardless of filter , `... where false`
   q="*" : Everything, apply filter if provided, `... where true` 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Searching `q="*"` now returns all documents.

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
Yes, added a new test `test_empty_lexical_query`

* **Related Python client changes** (link commit/PR here)

* **Related documentation changes** (link commit/PR here)

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: tests/tensor_search/integ_tests/test_search_unstructured.py [hunk 1]
```diff
@@ -1375,3 +1375,20 @@ def test_lexical_query_can_not_be_none(self):
                 with self.assertRaises(InvalidArgError):
                     res = tensor_search.search(text=None, config=self.config, index_name=self.default_text_index,
                                                search_method=SearchMethod.LEXICAL)
+                    
+    def test_empty_lexical_query(self):
+        tensor_search.add_documents(config=self.config,
+                                            add_docs_params=AddDocsParams(
+                                                index_name=self.default_text_index,
+                                                docs=[
```
Comment (vicilliar) [inline tests/tensor_search/integ_tests/test_search_unstructured.py]: Can we make a test that shows that multiple documents can be returned like this? Just to show the query will return all documents, not just 1.

**Review by vicilliar — CHANGES_REQUESTED** [review]

File: tests/tensor_search/integ_tests/test_search_unstructured.py (hunk 1 above)
Comment (RaynorChavez) [reply tests/tensor_search/integ_tests/test_search_unstructured.py]: done!

File: src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py [hunk 2]
```diff
@@ -234,6 +235,12 @@ def _get_score_modifiers(marqo_query: MarqoQuery) -> \
 
     def _to_vespa_lexical_query(self, marqo_query: MarqoLexicalQuery) -> Dict[str, Any]:
         def _get_lexical_search_term(marqo_query: MarqoLexicalQuery) -> str:
+
+            if not marqo_query.or_phrases and not marqo_query.and_phrases:
+                return 'false'
+            if marqo_query.or_phrases == ["*"] or marqo_query.and_phrases == ["*"]:
```
Comment (vicilliar) [inline src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py]: This will trigger on edge cases like:
`q='"exact" *'` or `q='"*" optional'`. Specifically, we only want the "fetch all" behavior to happen on `q='*'`. You may have to test this, but I think this means:
```
marqo_query.or_phrases == ["*"] and not marqo_query.and_phrases
```

File: tests/tensor_search/integ_tests/test_search_unstructured.py [hunk 3]
```diff
@@ -1375,3 +1375,42 @@ def test_lexical_query_can_not_be_none(self):
                 with self.assertRaises(InvalidArgError):
                     res = tensor_search.search(text=None, config=self.config, index_name=self.default_text_index,
                                                search_method=SearchMethod.LEXICAL)
+                                                    
+    def test_empty_lexical_query(self):
```
Comment (vicilliar) [inline tests/tensor_search/integ_tests/test_search_unstructured.py]: 1. Move these tests to `test_search_combined`. We're trying to move away from files specifically for structured/unstructured
2. separate the tests for `*` from the ones for `""`, with descriptive test names
3. Make test for wildcard with filter
4. Make test for the edge cases mentioned above (wildcard `or term` with text `and term`, vice verse)

**Review by vicilliar — CHANGES_REQUESTED** [review]

File: src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py (hunk 2 above)
Comment (RaynorChavez) [reply src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py]: yes you're right! That was my initial implementation as well, but changed it to the above. Will revert. Thanks Joshua

File: tests/tensor_search/integ_tests/test_search_unstructured.py (hunk 3 above)
Comment (RaynorChavez) [reply tests/tensor_search/integ_tests/test_search_unstructured.py]: created!

**Review by vicilliar — DISMISSED** [review]

File: src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py [hunk 4]
```diff
@@ -11,6 +11,9 @@
 from marqo.core.unstructured_vespa_index.unstructured_document import UnstructuredVespaDocument
 from marqo.core.vespa_index import VespaIndex
 from marqo.exceptions import InternalError, InvalidArgumentError
+from marqo.logging import get_logger
+
+logger = get_logger(__name__)
```
Comment (farshidz) [inline src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py]: Why did you add this? It isn't used

File: src/marqo/tensor_search/tensor_search.py [hunk 5]
```diff
@@ -1384,7 +1384,7 @@ def _lexical_search(
         raise api_exceptions.InvalidArgError(
             f"Query arg must be of type str! text arg is of type {type(text)}. "
             f"Query arg: {text}")
-
+
```
Comment (farshidz) [inline src/marqo/tensor_search/tensor_search.py]: Please make sure to format your code so extra whitespace it removed

File: tests/tensor_search/integ_tests/test_search_combined.py [hunk 6]
```diff
@@ -644,3 +642,82 @@ def test_search_vectoriseIsCalledWithEnableCacheTrue(self):
...
+
+    def test_empty_lexical_query(self):
+        for index in [self.structured_default_text_index, self.unstructured_default_text_index]:
+            with self.subTest(index=index.type):
+                tensor_search.add_documents(
+                    config=self.config,
+                    add_docs_params=AddDocsParams(
+                        index_name=index.name,
+                        docs=[
+                            {"_id": "1", "text": "document_1"},
```
Comment (farshidz) [inline tests/tensor_search/integ_tests/test_search_combined.py:659]: Try to make your test as simple and concise as possible. Having so many documents here doesn't add any value. I think 3-4 docs would suffice.

File: tests/tensor_search/integ_tests/test_search_combined.py [hunk 7]
```diff
@@ -644,3 +642,82 @@ def test_search_vectoriseIsCalledWithEnableCacheTrue(self):
...
+                    )
+                )
+
+                # Assert that no documents are returned for an empty query
+                res = tensor_search.search(text="", config=self.config, index_name=index.name,
+                                           search_method=SearchMethod.LEXICAL, result_count=10)
+                self.assertIn("hits", res)
+                self.assertEqual(0, len(res['hits']))
+
+    def test_wildcard_lexical_query_structured(self):
```
Comment (farshidz) [inline tests/tensor_search/integ_tests/test_search_combined.py]: A test description will help a lot. Specifically that you're expecting all results for * but other wildcards are interpreted literally and not supported

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py (hunk 4 above)
Comment (RaynorChavez) [reply src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py]: Removed! I just added it for future ease of use.

File: tests/tensor_search/integ_tests/test_search_combined.py (hunk 7 above)
Comment (RaynorChavez) [reply tests/tensor_search/integ_tests/test_search_combined.py]: Added!

File: tests/tensor_search/integ_tests/test_search_combined.py (hunk 6 above)
Comment (RaynorChavez) [reply tests/tensor_search/integ_tests/test_search_combined.py:659]: Revised to four documents!

**Review by farshidz — APPROVED** [review]
